src/modules/reports/reports.service.ts

```diff
@@ -0,0 +1,41 @@
+import { Injectable, BadRequestException } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { PatientEntity } from '../patient/models/patient.entity';
+import { AppointmentEntity } from '../appointment/models/appointment.entity';
+
+@Injectable()
+export class ReportsService {
+  constructor(
+    @InjectRepository(PatientEntity)
+    private patientRepository: Repository<PatientEntity>,
+    @InjectRepository(AppointmentEntity)
+    private appointmentRepository: Repository<AppointmentEntity>,
+  ) {}
+
+  async findBetweenDates(
+    startDateString: string,
+    endDateString: string,
+  ): Promise<any[]> {
+    try {
+      const startDate = new Date(startDateString);
+      const endDate = new Date(endDateString);
+      if (startDate > endDate) {
+        throw new BadRequestException(
+          'La fecha de inicio debe ser menor o igual a la fecha de fin',
```

Comment:
can u use constants and the examples as the another exceptions? this must be in english too